openhtf/__init__.py

```diff
@@ -83,10 +87,22 @@ class TestPhaseInfo(mutablerecords.Record(
   """
 
   @classmethod
-  def WrapOrReturn(cls, func):
+  def WrapOrCopy(cls, func):
+    """Return a new TestPhaseInfo from the given function or instance.
+    
+    We want to return a new copy so that you can reuse a phase with different
+    options, plugs, measurements, etc.
+    
+    Args:
+      func: A phase function or TestPhaseInfo instance.
+    
+    Returns:
+      A new TestPhaseInfo object.
+    """
     if not isinstance(func, cls):
-      func = cls(func, inspect.getsource(func))
-    return func
+      return cls(func, inspect.getsource(func))
+    # We want to copy so that a phase can be reused
+    return copy.copy(func)
```

Comment:
deepcopy is tricky because you can't say how many levels. I want to copy the _list_ of plugs, but I don't want to copy the actual plug classes as that can do some nasty stuff depending on the class. Though you're right, this only makes a new Record with the same stuff inside. Any suggestions other than 'deepcopy and we'll see when it fails' ?